tanner/utils/patterns.py

```diff
@@ -12,6 +12,7 @@
 PHP_CODE_INJECTION = re.compile(r'.*(;)*(echo|system|print|phpinfo)(\(.*\)).*')
 PHP_OBJECT_INJECTION = re.compile(r'(^|;|{|})O:[0-9]+:')
 CRLF_ATTACK = re.compile(r'.*(\r\n).*')
+XXE_INJECTION = re.compile(r'(^<\?xml)(\s)(version=.*)(\s)(encoding=.*)(\?>).*')
```

Comment:
Please consider these things for regex
1. Check whether any text before keyword `<?xml` make it valid.
2. `\s` can occur multiple time, not just `\s` maybe various fields(other than `version/encoding`) can be placed here
3. I think it possible to avoid `version/encoding`.
4. Also check whether we can start directly from`<!DOCTYPE` and avoid `<?xml ... ?>`

Reply:
Yes, we can start directly from `<!DOCTYPE` and the code would work but I thought we should follow the DTD structure.

Reply:
I think we should consider all the possibilities so relax the regex.

Reply:
Okay 😅 .